clippy_lints/src/panic.rs

```diff
@@ -45,6 +46,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
             is_direct_expn_of(cx, params[0].span, "panic").is_some(),
             let LitKind::Str(ref string, _) = lit.node,
             let Some(par) = string.as_str().find('{'),
+            !string.as_str()[par + 1..min(par + 2, string.as_str().len())].contains('{'),
```

Comment:
`string[par]` is `{`, but this condition tries to slice the next char (par+1..par+2), which might be a start of multibyte character.